packages/ra-core/src/dataProvider/useDataProvider.ts

```diff
@@ -143,6 +143,7 @@ const useDataProvider = (): DataProviderProxy => {
                         undoable = false,
                         onSuccess = undefined,
                         onFailure = undefined,
+                        mutationMode = 'pessimistic',
```

Comment:
You can probably initialize `mutationMode` using the `undoable` option here and avoid referencing `undoable` below

```suggestion
                        mutationMode = undoable ? 'undoable' : 'pessimistic',
```